drake/core/test/vector_test.cc

```diff
@@ -0,0 +1,142 @@
+  test = 2 * abc;
+
+  EXPECT_EQ(test.first().theta, 2.0);
+  EXPECT_EQ(test.first().thetadot, 4.0);
+  EXPECT_EQ(test.second().tau, 6.0);
+}
+
+// Tests the ability to use a CombinedVectorUtil
+TEST(VectorTest, CombinedVectorUtil) {
+  Eigen::Vector3d abc;
+  abc << 1, 2, 3;
+
+  CombinedVectorUtil<PendulumState, PendulumInput>::type<double> test(abc);
+  test = 2 * abc;
+  EXPECT_EQ(test.first().theta, 2.0);
+  EXPECT_EQ(test.first().thetadot, 4.0);
+  EXPECT_EQ(test.second().tau, 6.0);
+}
+
+// Verify that combining a vector with an unused or empty vector returns the
+// original type
+TEST(VectorTest, CombineVectorCornerCases) {
+  CombinedVectorUtil<PendulumState, NullVector>::type<double> test1;
+  EXPECT_TRUE((is_same<PendulumState<double>, decltype(test1)>::value))
+      << "combined vector builder returned " +
+             static_cast<string>(typeid(test1).name());
+
+  CombinedVectorUtil<NullVector, PendulumState>::type<double> test2;
+  EXPECT_TRUE((is_same<PendulumState<double>, decltype(test2)>::value))
+      << "combined vector builder returned " +
+             static_cast<string>(typeid(test2).name());
+}
+
+// Tests the RowsAtCompileTime
+TEST(VectorTest, RowsAtCompileTime) {
+  EXPECT_EQ((Eigen::Matrix<double, 2, 1>::RowsAtCompileTime), 2)
+      << "failed to evaluate RowsAtCompileTime";
+}
+
+// Tests the InputOutputRelation. Verify that linear is a polynomial.
+TEST(VectorTest, InputOutputRelationLinearIsPolynomial) {
+  EXPECT_TRUE((InputOutputRelation::isA(InputOutputRelation::Form::LINEAR,
+                                        InputOutputRelation::Form::POLYNOMIAL)))
+      << "linear is polynomial";
+}
+
+// Tests the InputOutputRelation. Verify that zero is arbitrary.
+TEST(VectorTest, InputOutputRelationZeroIsArbitrary) {
+  EXPECT_TRUE((InputOutputRelation::isA(InputOutputRelation::Form::ZERO,
+                                        InputOutputRelation::Form::ARBITRARY)))
+      << "zero is arbitrary";
+}
+
+// Verifies that the least common ancestor of the I/O relations
+// AFFINE, LINEAR, AND POLYNOMIAL is polynomial.
+TEST(VectorTest, InputOutputRelationLeastCommonAncestor) {
+  EXPECT_TRUE((
+      InputOutputRelation::leastCommonAncestor(
+          {InputOutputRelation::Form::AFFINE, InputOutputRelation::Form::LINEAR,
+           InputOutputRelation::Form::POLYNOMIAL}) ==
+      InputOutputRelation::Form::POLYNOMIAL))
+      << "least common ancestor should be polynomial";
+}
+
+// Verifies that compositions of I/O relations are as expected
+TEST(VectorTest, InputOutputRelationCompositionTests) {
+  InputOutputRelation g(InputOutputRelation::Form::LINEAR);
+  InputOutputRelation f(InputOutputRelation::Form::POLYNOMIAL);
+
+  EXPECT_EQ(InputOutputRelation::composeWith(g, f).form,
+              InputOutputRelation::Form::POLYNOMIAL);
+
+  EXPECT_EQ(InputOutputRelation::composeWith(f, g).form,
+              InputOutputRelation::Form::POLYNOMIAL);
```

Comment:
@david-german-tri: I just ran clang-format and noticed it may be violating the style guide's specification on the order of includes. Here is an example change made by clang-format:

![screen shot 2016-03-31 at 3 00 24 pm](https://cloud.githubusercontent.com/assets/1388098/14187584/c9ff1c8a-f751-11e5-9d50-1bf48ab2b816.png)

Shouldn't `gtest.h` be considered "other libraries' .h" meaning it should go before drake's header files?